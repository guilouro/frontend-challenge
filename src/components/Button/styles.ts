import styled, { css } from 'styled-components'

export const Wrapper = styled.button`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    background-color: ${theme.colors.base};
    border-radius: ${theme.border.radius};
    border: 0;
    color: #fff;
    cursor: pointer;
    font-family: ${theme.font.family};
    font-size: ${theme.font.sizes.small};
    min-height: 4rem;
    padding: ${theme.spacings.xxsmall};
    text-decoration: none;
    transition: ${theme.transition.fast} background-color;

    &:hover {
      background-color: ${theme.colors.baseDark};
    }

    &:disabled {
      background-color: ${theme.colors.textLight};
    }
  `}
`
