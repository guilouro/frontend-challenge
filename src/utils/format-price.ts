export default function formatPrice(price: number | bigint): string {
  return new Intl.NumberFormat('de-DE', {
    style: 'currency',
    currency: 'EUR'
  }).format(price)
}
