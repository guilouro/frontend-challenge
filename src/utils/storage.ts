import { CartObject } from 'hooks/use-cart'

const APP_KEY = 'BARKYN_CART'

export function getStorageItem() {
  if (typeof window === 'undefined') return

  const data = window.localStorage.getItem(APP_KEY)
  return JSON.parse(data || '[]')
}

export function setStorageItem(value: CartObject) {
  if (typeof window === 'undefined') return

  const data = JSON.stringify(value)
  return window.localStorage.setItem(APP_KEY, data)
}
