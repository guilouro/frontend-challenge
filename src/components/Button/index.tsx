import { forwardRef, AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react'

import * as S from './styles'

type ButtonTypes =
  | AnchorHTMLAttributes<HTMLAnchorElement>
  | ButtonHTMLAttributes<HTMLButtonElement>

export type ButtonProps = {
  as?: React.ElementType
} & ButtonTypes

const Button: React.ForwardRefRenderFunction<ButtonTypes, ButtonProps> = (
  { children, ...props },
  ref
) => (
  <S.Wrapper {...props} ref={ref}>
    <span>{children}</span>
  </S.Wrapper>
)

export default forwardRef(Button)
