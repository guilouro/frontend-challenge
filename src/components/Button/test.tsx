import { render, screen } from 'utils/test-utils'

import Button from '.'

describe('<Button />', () => {
  it('should render as button', () => {
    const { container } = render(<Button>Click here</Button>)
    expect(
      screen.getByRole('button', { name: /click here/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render as link', () => {
    render(
      <Button as="a" href="/checkout">
        Click Here
      </Button>
    )
    expect(screen.getByRole('link', { name: /click here/i })).toHaveAttribute(
      'href',
      '/checkout'
    )
  })
})
