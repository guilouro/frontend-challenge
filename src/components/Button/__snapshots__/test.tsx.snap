// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render as button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  background-color: #5f2eff;
  border-radius: 1.6rem;
  border: 0;
  color: #fff;
  cursor: pointer;
  font-family: Azo sans,sans-serif;
  font-size: 1.4rem;
  min-height: 4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 0.2s ease-in-out background-color;
  transition: 0.2s ease-in-out background-color;
}

.c0:hover {
  background-color: #5400db;
}

.c0:disabled {
  background-color: #6c6b76;
}

<button
  class="c0"
>
  <span>
    Click here
  </span>
</button>
`
